wicker/core/storage.py

```diff
@@ -79,9 +102,40 @@ def fetch_file(self, input_path: str, local_prefix: str, timeout_seconds: int =
         :param timeout_seconds: number of seconds till timing out on waiting for the file to be downloaded
         :return: local path to the file on the local file system
         """
-        input_path = os.path.join(input_path, os.path.basename(local_prefix))
-        self.download_with_retries(input_path, local_prefix)
-        return local_prefix
+        local_full_path = os.path.join(local_prefix, os.path.basename(input_path))
+
+        # for now assume if local file exists don't re-download, need to check file sum longer term
+        lock_path = local_full_path + ".lock"
+        success_marker = local_full_path + ".success"
+        while not os.path.isfile(success_marker):
+            with SimpleUnixFileLock(lock_path, timeout_seconds=timeout_seconds):
```

Comment:
It looks like this duplicates the same from `S3DataStorage`. let's consolidate since the only difference seems to be the way we download it.

Reply:
@pickles-bread-and-butter +1 for a helper function that can be used for both classes.

Reply:
Done